benchmarks/publish-subscribe/src/main.rs

```diff
@@ -79,16 +80,42 @@ fn perform_benchmark<T: Service>() {
         t2.join().expect("thread failure");
         let stop = start.elapsed().expect("failed to measure time");
         println!(
-            "{} ::: Time: {}, Latency: {} ns",
+            "{} ::: Iterations: {}, Time: {}, Latency: {} ns",
             std::any::type_name::<T>(),
+            iterations,
             stop.as_secs_f64(),
-            stop.as_nanos() / (ITERATIONS as u128 * 2)
+            stop.as_nanos() / (iterations as u128 * 2)
         );
     });
 }
 
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Args {
+    /// Number of iterations the A --> B --> A communication is repeated
+    #[clap(short, long, default_value_t = ITERATIONS)]
+    iterations: u64,
+    /// Run benchmark for every service setup
+    #[clap(short, long, default_value_t = false)]
+    bench_all: bool,
+    /// Run benchmark for the IPC zero copy setup
+    #[clap(short, long, default_value_t = false)]
+    bench_zero_copy: bool,
+    /// Run benchmark for the process local setup
+    #[clap(short, long, default_value_t = false)]
+    bench_process_local: bool,
+}
+
 fn main() {
+    let args = Args::parse();
+
     set_log_level(iceoryx2_bb_log::LogLevel::Error);
-    perform_benchmark::<zero_copy::Service>();
-    perform_benchmark::<process_local::Service>();
+
+    if args.bench_zero_copy || args.bench_all {
+        perform_benchmark::<zero_copy::Service>(args.iterations);
+    }
+
+    if args.bench_process_local || args.bench_all {
+        perform_benchmark::<process_local::Service>(args.iterations);
+    }
 }
```

Comment:
Can you print something when no bench option is set? Like "Please use either '--bench_all' or ... "